source/features/filter-pr-by-build-status.tsx

```diff
@@ -0,0 +1,70 @@
+import React from 'dom-chef';
+import select from 'select-dom';
+import {checkInline} from '../libs/icons';
+import features from '../libs/features';
+import {getOwnerAndRepo} from '../libs/utils';
+
+function init() {
+	const filterAssignee = select('.table-list-filters > .table-list-header-toggle:last-child > details:nth-child(7)');
+
+	filterAssignee.parentNode.insertBefore(
+		<details class="details-reset details-overlay float-left select-menu">
+			<summary class="btn-link select-menu-button" aria-haspopup="menu">
+				Status&nbsp;
+			</summary>
+			<details-menu class="select-menu-modal position-absolute right-0" style={{zIndex: 99}} role="menu">
+				<div class="select-menu-header">
+					<span class="select-menu-title">Filter by Status</span>
+				</div>
+				<div class="select-menu-list">
+					{getStatusDropDown()}
+				</div>
+			</details-menu>
+		</details>,
```

Comment:
Can you please explain a bit more on this? I'll have to deep clone the node. Right? Because passing false will only shallow clone the node, 
`<details class="details-reset details-overlay select-menu d-inline-block">` which I don't think would be any helpful.  

Also, Even after deep-cloning, I'll have to change the content based on some `if`s as after cloning I'm getting 5 children nodes among which three are just text-nodes, So it will be little dirty. 

```
    const statusFilterNode = filterAssignee.cloneNode(true);
	filterAssignee.before(statusFilterNode);

	const summary = statusFilterNode.childNodes;

	for (const childNode of statusFilterNode.childNodes) {
		if (childNode.constructor === HTMLElement) {
           if (childNode.tagName === 'SUMMARY') {
              // Manipulate summary
           }

           if (childNode.tagName === 'DETAILS-MENU') {
              // Manipulate details-menu
           }
		}
	}
``` 
or I just hard-code the index as `statusFilterNode[1]` and `statusFilterNode[3]`? Which would be clean and maintainable from your POV?

Reply:
Indeed, I meant `cloneNode(true)` to include all children. 

Why 5 places? I see only 2 custom strings and they could be replaced with:

```is
const filter = select('......').cloneNode(true);
select('summary', filter).textContent = 'Status\xA0';
select('.select-menu-title', filter).textContent = 'Filter by Status';
```

Reply:
I should have been a little more clear. I was following the above index based access approach as it was returning me the `HTMLElement` type whereas the filter here is of type `Node` so `select-dom` was complaining.
```
Argument of type 'Node' is not assignable to parameter of type 'BaseElement'.
```

For now, avoided as problem by casting it as 
```
select('...').cloneNode(true) as HTMLElement;
```
Let me know if we want to avoid this.